asb-ballerina/commons.bal

```diff
@@ -46,12 +38,77 @@ public enum receiveModes {
 #                      Endpoint=sb://namespace_DNS_Name;EntityPath=EVENT_HUB_NAME;
 #                      SharedAccessKeyName=SHARED_ACCESS_KEY_NAME;SharedAccessKey=SHARED_ACCESS_KEY or  
 #                      Endpoint=sb://namespace_DNS_Name;EntityPath=EVENT_HUB_NAME;
-#                      SharedAccessSignatureToken=SHARED_ACCESS_SIGNATURE_TOKEN
-@display {label: "Connection Config"}
-public type AsbConnectionConfiguration record {|
-    @display {label: "Connection String"}
+#                      SharedAccessSignatureToken=SHARED_ACCESS_SIGNATURE_TOKEN 
+# + entityConfig -  This field holds the configuration details of either a topic or a queue. The type of the entity is 
+#                   determined by the entityType field. The actual configuration details are stored in either a 
+#                   TopicSubsConfig or a QueueConfig record  
+# + entityType - This field is an enumeration (EntityType) that indicates whether the entity is a topic or a queue  
+# + receiveMode - This field holds the receive modes(RECEIVE_AND_DELETE/PEEK_LOCK) for the connection. The receive mode determines how messages are 
+#                 retrieved from the entity. The default value is PEEK_LOCK
+@display {label: "Receiver Connection Config"}
+public type ASBServiceReceiverConfig record {
+    @display {label: "Entity Type"}
+    EntityType entityType;
+    @display {label: "Entity Configuration"}
+    TopicSubsConfig|QueueConfig entityConfig;
+    @display {label: "ConnectionString"}
+    string connectionString;
+    @display {label: "Receive Mode"}
+    ReceiveMode receiveMode = PEEK_LOCK;
+};
+
+# This record holds the configuration details of a topic and its associated subscription in Azure Service Bus
+#
+# + topicName - A string field that holds the name of the topic  
+# + subscriptionName - A string field that holds the name of the subscription associated with the topic
+@display {label: "Topic/Subscriptions Configurations"}
+public  type TopicSubsConfig record {
+   @display {label: "Topic Name"}
+   string topicName;
+   @display {label: "Subscription Name"}
+   string subscriptionName; 
```

Comment:
The subscriptionName is mandatory thing to provide?

Reply:
Yes, It is for Receiver. Previous version , it was provided along with the entity path like this : topicName/Subscription/subscriptionName, However latest 7.13.1 SDK needs them as separate parameters.